server/game/cards/08.3-Km/TheSkahazadhan.js

```diff
@@ -0,0 +1,25 @@
+const DrawCard = require('../../drawcard.js');
+
+class TheSkahazadhan extends DrawCard {
+    setupCardAbilities(ability) {
+        this.reaction({
+            when: {
+                onPlotsRevealed: () => true
```

Comment:
This isn't quite correct. The triggering condition is "when **you** reveal a plot card." As it stands, this will trigger when your opponent reveals a plot using Rains. See Caswell's Keep for the proper condition.